chromium/popup.html

```diff
@@ -1,74 +1,78 @@
+          <p i18n="chrome_always_https_for_host"></p>
+          <label for="new-rule-host" i18n="chrome_host"></label><br>
+          <input size="50" id="new-rule-host" type="text" disabled><br>
+          <div id="new-rule-regular-text">
+            <a href="javascript:void 0" id="new-rule-show-advanced-link" i18n="chrome_show_advanced"></a><br>
+          </div>
+          <div id="new-rule-advanced" style="display: none">
+            <a href="javascript:void 0" id="new-rule-hide-advanced-link" i18n="chrome_hide_advanced"></a><br>
+            <br>
+            <label for="new-rule-name" i18n="chrome_rule_name"></label><br>
+            <input size="50" id="new-rule-name" type="text"><br>
+            <label for="new-rule-regex" i18n="chrome_regex"></label><br>
+            <input size="50" id="new-rule-regex" type="text"><br>
+            <label for="new-rule-redirect" i18n="chrome_redirect_to"></label><br>
+            <input size="50" id="new-rule-redirect" type="text"><br>
+          </div>
+          <button id="add-new-rule-button" i18n="chrome_add_new_rule"></button><br>
+          <button id="cancel-new-rule" i18n="chrome_status_cancel_button"></button>
         </div>
-    </section>
-    <section id="StableRules" class="rules">
+      </section>
+
+      <section id="StableRules" class="rules">
         <h3 i18n="chrome_stable_rules"></h3>
         <p class="description" i18n="chrome_stable_rules_description"></p>
-    </section>
-    <section id="UnstableRules" class="rules">
+      </section>
+
+      <section id="UnstableRules" class="rules">
         <h3 i18n="chrome_experimental_rules"></h3>
         <p class="description" i18n="chrome_experimental_rules_description"></p>
-    </section>
-</div>
+      </section>
+    </div>
 
-<footer>
-    <a id="viewAllRules" href="https://www.eff.org/https-everywhere/atlas" target="_blank" i18n="menu_viewAllRules"></a><br><br>
-    <a id="aboutTitle" href="https://www.eff.org/https-everywhere" target="_blank" i18n="about_title"></a><br><br>
-    <a id="donateEFF" href="https://supporters.eff.org/donate/support-https-everywhere" target="_blank" i18n="menu_donate_eff_imperative"></a>
-    <p><small>(<span i18n="about_version">Version</span>: <span id="current-version"></span>)</small></p>
-</footer>
+    <footer>
+      <a id="viewAllRules" href="https://www.eff.org/https-everywhere/atlas" target="_blank" i18n="menu_viewAllRules"></a><br>
+      <br>
+      <a id="aboutTitle" href="https://www.eff.org/https-everywhere" target="_blank" i18n="about_title"></a><br>
+      <br>
+      <a id="donateEFF" href="https://supporters.eff.org/donate/support-https-everywhere" target="_blank" i18n="menu_donate_eff_imperative"></a>
+      <p><small>(<span i18n="about_version">Version</span>: <span id="current-version"></span>)</small></p>
+    </footer>
 
-</body>
+    <script src="popup.js"></script>
+    <script src="translation.js"></script>
+    <script src="send-message.js"></script>
```

Comment:
There's no reason to have these within the body.  Code referring to the DOM in these files are wrapped in

```js
document.addEventListener("DOMContentLoaded", () => ...);
```

Reply:
@Hainish It's recommended to always place `script` tags at the bottom of `body` tag.